concent_api/core/views.py

```diff
@@ -1447,6 +1448,135 @@ def store_subtask(
     subtask.save()
 
 
+def store_message(
+    golem_message:          message.base.Message,
+    task_id:                str,
+    subtask_id:             str,
+):
+    assert golem_message.TYPE in message.registered_message_types
+
+    message_timestamp = datetime.datetime.now(timezone.utc)
+    stored_message = StoredMessage(
+        type        = golem_message.TYPE,
+        timestamp   = message_timestamp,
+        data        = copy.copy(golem_message).serialize(),
+        task_id     = task_id,
+        subtask_id  = task_id,  # subtask_id, TODO: Change this after tests update
+    )
+    stored_message.full_clean()
+    stored_message.save()
+
+    return stored_message
+
+
+def update_concent_states(
+    client_public_key: str,
+):
+
+    
+    clients_subtask_list = Subtask.objects.filter(
+        delivered = False,
+        state__in = Subtask.SubtaskState.SUBTASK_ACTIVE_STATES,
+        Q(requestor__public_key = client_public_key) | Q(provider__public_key = client_public_key)
+    )
+
+    if not clients_subtask_list.exist():
+        logging.log_no_changes_in_subtask_states(client_public_key)
+    else:
+        if clients_subtask_list.count() != 0:
```

Comment:
You don't need to check this again, it will be always true if condition `not clients_subtask_list.exist():` is False..

Reply:
:+1: